### PR TITLE
Enable download funnel experiment on stage and prod

### LINKED ISSUE #10753: Enable download funnel experiment on stage and prod
Everything is ready for us to run the experiment, so we can enable it on stage next Tuesday and on prod next Thursday.

### PR DESCRIPTION
**Pull request by bobsilverberg**
Fixes #10753

### CODE REVIEW

File: config/test.js [hunk 1]
```diff
@@ -10,4 +10,9 @@ module.exports = {
   enableRequestID: false,
 
   mozillaUserId: 1337,
+
+  // Disable all experiments by default.
+  experiments: {
+    '20210531_amo_download_funnel_experiment': false,
```
Comment (willdurand) [inline config/test.js:16]: Can we import `config/default.js` and spread `config.experiments` here instead? (if we really want to disable all experiments by default)

Comment (bobsilverberg) [reply config/test.js]: I'm not sure I follow. Do you mean to get the keys from `config/default.js` and explicitly set each one to `false`?

Comment (willdurand) [reply config/test.js]: Yes:

```js
...Object.keys(defaultConfig.experiments).reduce((acc, key) => {
  return { ...acc, [key]: false };
}, {});
```

Comment (bobsilverberg) [reply config/test.js]: Right, I thought that might have been what you meant. Do you think that is a better approach, or the mocking approach that is now present in this patch? Or maybe something else?

File: tests/unit/amo/components/TestGetFirefoxBanner.js [hunk 2]
```diff
@@ -128,7 +133,7 @@ describe(__filename, () => {
     });
 
     it('sets the href on the button with the expected utm params when there is no variant', () => {
-      const root = render({ store });
+      const root = render({ store, variant: undefined });
```
Comment (willdurand) [inline tests/unit/amo/components/TestGetFirefoxBanner.js:136]: Why is this needed now?

Comment (bobsilverberg) [reply tests/unit/amo/components/TestGetFirefoxBanner.js:136]: With the experiment enabled globally (which includes in the tests), the `withExperiment` HOC is now running and calculating a `variant` just as it is meant to. This means that if we do not manually override `variant` with `undefined` in `render`, then the HOC assigns a variant and this test is no longer testing the case of "when there is no variant". This was passing before, without the explicit override of `variant`, because the experiment was not enabled, and therefore the HOC returned `undefined` for `variant`.

File: tests/unit/amo/components/TestGetFirefoxBanner.js [hunk 3]
```diff
@@ -29,6 +29,11 @@ import {
   shallowUntilTarget,
   userAgents,
 } from 'tests/unit/helpers';
+// eslint-disable-next-line no-unused-vars
+import tracking from 'amo/tracking';
```
Comment (willdurand) [inline tests/unit/amo/components/TestGetFirefoxBanner.js:33]: Is this required to make `jest.mock` work?

**Comment by willdurand** [comment]
I think it looks quite good but I have mixed feelings about `jest.mock()`

File: tests/unit/amo/components/TestGetFirefoxBanner.js (hunk 3 above)
Comment (bobsilverberg) [reply tests/unit/amo/components/TestGetFirefoxBanner.js:33]: Yes, exactly.

Comment (bobsilverberg) [reply tests/unit/amo/components/TestGetFirefoxBanner.js:33]: Actually, we don't need it. I never tested without it, and just did, and I guess it's not needed at all. I'll remove these imports and comments.
